### PR TITLE
Dependency cleanup

### LINKED ISSUE #120: Missing `estraverse` dependency
Hi, I tried this project today and at first glance it seems to work very well. However, when I tried installing the dependencies of the example project with [pnpm](https://pnpm.js.org/), I got this error on compilation:
```
/src/index.js
Line 0:  Parsing error: Cannot find module 'estraverse'
```
However, there is no such line in `examples/src/index.js`.
This probably means one of the (deep) dependencies is missing `estraverse` as a dependency in their `package.json`, but they're relying on it being installed in the `node_modules` folder because it's a dependency of one of their dependencies. According to `npm`'s specification, only direct dependencies have to be installed in the `node_modules` folder, which is exactly what `pnpm` does.

Sadly, the error message doesn't give me any idea of what package might be missing the `estraverse` dependency, which is where I hope someone can help out.

For anyone who runs into this and doesn't know, the issue can be resolved by using `pnpm`'s `--shamefully-flatten` flag.

### PR DESCRIPTION
**Pull request by cdfa**
Fixes #120, #119 and #77 

Changes:
* Update dependencies
* move prop-types dependency from example to lib. It's weird to put this as a peer dependency, because we only import it in the library and a user may choose to not use prop-types or something different.
* remove `react-dom` dependency from library (still used in example). It doesn't look like this is actually used.
* add eslint-import-resolver-node (doesn't result in build error, but eslint complains without it)
* move all babel config to `.babel.rc`
* Add all installed plugins to babel config 

Notes
* warning about rollup postcss plugin using the onwrite hook: https://github.com/egoist/rollup-plugin-postcss/pull/148
* the specific versions babel-eslint and eslint are required by the latest version of react-scripts.
* missing peer dependency from `rollup-plugin-peer-deps-external` doesn't seem to matter. [Issue](https://github.com/Updater/rollup-plugin-peer-deps-external/issues/10)

### CODE REVIEW

**Comment by cdfa** [comment]
Something is broken with the testing. When i try to run them myself it eats up way to much ram and the travis ci log says it's finding `eslint 4.18.2`, which is not the one specified in the new, or old root `package.json`. I don't get the error myself and it's annoying me that `npm` looks for packages up the directory tree, preventing the root project to use a different `npm` version than `react-scripts`. That's just a flawed design from the start.

**Comment by paulmelnikow** [comment]
Hey! I've been meaning to dip my toe in to this library and joined some discussions but realized I wasn't following, and missed this before.

There's a lot happening in this PR which makes it a bit difficult to review. What do you think about picking out little pieces into new PRs and trying to get them merged?

I don't have merge access (though @transitive-bullshit started a convo about that https://github.com/transitive-bullshit/create-react-library/issues/104#issuecomment-444359803). However I'd be happy to review and 👍 pieces of this, and I think it would expedite this update.

**Comment by paulmelnikow** [comment]
> * move prop-types dependency from example to lib. It's weird to put this as a peer dependency, because we only import it in the library and a user may choose to not use prop-types or something different.

Agreed this is a good idea. When I use `prop-types` in my libraries, I make it a dependency. I think the counterargument is trying to avoid multiple versions of prop-types being installed, though I tend to trade off in favor of simplicity by making it a dep. It's not very big.

**Comment by paulmelnikow** [comment]
> * remove `react-dom` dependency from library (still used in example). It doesn't look like this is actually used.

👍

File: template/default/package.json [hunk 1]
```diff
@@ -45,32 +42,35 @@
...
+    "rollup-plugin-postcss": "^1.6.3",
+    "rollup-plugin-url": "^2.1.0"
   },
   "files": [
     "dist"
-  ]
+  ],
+  "dependencies": {
+    "prop-types": "^15.6.2"
+  }
```
Comment (paulmelnikow) [inline template/default/package.json]: Could you move this up before `peerDependencies` so it's more noticeable?

**Comment by paulmelnikow** [comment]
> * Add all installed plugins to babel config

This makes sense to me.

**Comment by cdfa** [comment]
> There's a lot happening in this PR which makes it a bit difficult to review. What do you think about picking out little pieces into new PRs and trying to get them merged?

Yeah, I realized there's too much happening when I submitted the pull request, but I was quite fed up with these "chores" so I didn't split it up. I'll move a few things from my "change log" in my initial comments to notes for a start.

Al least it's good to see you agree with most of the changes though.

Do you have any idea why the CI tests are failing? I'm not really willing to continue working on this if I don't know what I'm doing wrong there. Does everything work fine with you? Please let me know.

**Comment by transitive-bullshit** [comment]
Thanks @cdfa.

I finally got around to updating CRL and releasing `v3.0.0`. Please feel free to follow-up with an updated PR and sorry for taking so long to get back to you.

**Pull request closed without merging**